js/mirador_viewer.js

```diff
@@ -1,57 +1,30 @@
-/*jslint browser: true*/
-/*global Mirador, textOverlayPlugin, Drupal*/
+/*jslint browser: true, esversion: 6 */
+/*global Mirador, Drupal, once*/
 /**
  * @file
  * Displays Mirador viewer.
  */
-(function ($, Drupal) {
+(function (Drupal, once) {
     'use strict';
 
-    /**
-     * If initialized.
-     * @type {boolean}
-     */
-    var initialized;
-    /**
-     * Unique HTML id.
-     * @type {string}
-     */
-    var base;
-
-    function init(context,settings){
-        if (!initialized){
-          initialized = true;
-
-          /*
-           Mirador
-           */
-          var miradorInstance = Mirador.viewer({
-            "id": base,
-            "manifests": {
-              [settings.iiif_manifest_url]: {provider: "Islandora"}
-            },
-            "window": settings.mirador_window_settings,
-            "windows": [
-                    {
-                        "manifestId": settings.iiif_manifest_url,
-                        "thumbnailNavigationPosition": 'far-bottom'
-                    }
-                ]
-            }, window.miradorPlugins);
-        }
-    }
-    Drupal.Mirador = Drupal.Mirador || {};
-
     /**
      * Initialize the Mirador Viewer.
      */
     Drupal.behaviors.Mirador = {
         attach: function (context, settings) {
-            base = settings.mirador_view_id;
-            init(context,settings);
+            Object.entries(settings.mirador.viewers).forEach(entry => {
+              const [base, values] = entry;
+              once('mirador-viewer', base, context).forEach(() =>
+                Mirador.viewer(values, window.miradorPlugins || {})
```

Comment:
The `window.miradorPlugins` thing here seems like we could have saner expectations? Like, have the integration itself include it itself, so there's not this arbitrary JS extension point?

Reply:
With something like https://github.com/roblib/mirador-integration-islandora/pull/1, could possibly avoid having to touch `window.miradorPlugins` here? Could just become:

```suggestion
                Mirador.viewer(values)
```